src/main/java/com/altenheim/kalender/controller/logicController/DateSuggestionController.java

```diff
@@ -0,0 +1,40 @@
+package com.altenheim.kalender.controller.logicController;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import com.altenheim.kalender.interfaces.IDateSuggestionController;
+import com.calendarfx.model.Entry;
+
+public class DateSuggestionController implements IDateSuggestionController 
+{
+    public Entry<String> getDateSuggestionFromEntryList(ArrayList<Entry<?>> input, LocalDateTime startSearchDateTime, int dateLenght) 
+    {
+        if (input == null || input.isEmpty() || input.get(input.size()-1).getEndAsLocalDateTime().isBefore(startSearchDateTime.plusMinutes((long) dateLenght + 1))) 
+            return null;
+
+        var runNumber = 0;
+        while (true) 
+        {
+            if (runNumber > input.size()-1)
+                return null;
+            
+            if (input.get(runNumber).getStartAsLocalDateTime().isAfter(startSearchDateTime))
+                startSearchDateTime = input.get(runNumber).getStartAsLocalDateTime();
```

Comment:
Ouch :-) Du manipulierst den Parameter. Stell dir vor ich nutze mein LocalDateTime Objekt in der Schleife die diese Methode aufruft nach dem ersten Durchlauf weiter - es hat dann andere Werte. Da deine Methode das nicht "weiß" (egal ob es dann wirklich so ist oder nicht) sollte man übergebene Objekte nie manipulieren. Es gibt Sonderfälle aber eigentlich ist es common Sense (zu Zeiten von C ging es oft nicht anders, aber in der OOP schon). also: Werte anfangs an lokales Objekt und mit dem dann rumspielen.